twisted/pydoctor
# [Fixes #248] Non zero exit code on warnings.

Scope
======

This adds a '-W' CLI option to have pydoctor exit with non zero code on warnings.

Tries to fix #248 

Changes
=======

Use warning when guessing the project name. This was done to simplify testing.

Add sphinx doc and apidocs tox env for easy manual / CI testings.

I am still using the private `_warning` method as I think that it should be made public in a separate PR.


Drive by changes
==============

I have fixed many of the sphinx warnings.

I have added the documentation build process to static checks. It is red, but I plan to fix it with the on the fly generation of help output.


How to test
=========

Change settings in tox.ini and run `tox -e apidocs` and see the results.
